packages/asset-swapper/src/utils/market_operation_utils/path_optimizer.ts

```diff
@@ -303,7 +303,7 @@ function findRoutesAndCreateOptimalPath(
 
         // TODO(kimpers): remove once we have solved the rounding/precision loss issues in the Rust router
         const maxSampledOutput = BigNumber.max(...routeSamples.map(s => s.output));
-        // Scale output by scale factor but never go above the largest sample (unknown liquidity) or below 1 base unit (unfillable)
+        // Scale output by scale factor but never go above the largest sample in sell quotes (unknown liquidity)  or below 1 base unit (unfillable)
```

Comment:
Do we need to worry about the output for buys at all? 😂 My understanding is that's the amount paid by the takers, and they review the quotes before they pay.

Reply:
I think you're right 👍 , but I wonder if we can end up in a situation where if we scale buy output downwards we may buy less than the `buyAmount` which would cause the trade to revert?

Reply:
Seems like the same scale factor applies to the `routeInput`, so the `buyAmount` is also scaled downwards in this case? But I could be wrong.

Reply:
yeah, and the price should not get worse for buying a smaller amount. So it should be safe 👍